treq/auth.py

```diff
@@ -26,6 +27,31 @@ def request(self, method, uri, headers=None, bodyProducer=None):
             method, uri, headers=headers, bodyProducer=bodyProducer)
 
 
+class _PinToFirstHostAgent(object):
+    """
+    An {twisted.web.iweb.IAgent} implementing object that takes two
+    agents, using the first as when the current request's host name
+    matches first request's host name, and the second when it does
+    not.
+    """
+
+    def __init__(self, first_agent, second_agent):
+        self._first_agent = first_agent
```

Comment:
I'm inclined to suggest Twisted coding-standard names rather than pep8, which seems to be the standard in this repo https://github.com/twisted/treq/blob/master/src/treq/testing.py#L45-L46 and is officially the standard in Klein (despite a few lapses there)